### DISCUSSION TITLE
NITRO ENGRDA doesn't work

### DISCUSSION THREAD

**Discussion by sherrick01821** [mdaus/nitro#394] [Q&A]
I have been unable to add an ENGRDA TRE successfully with Nitro. It complains either the ENGLLB[0] doesn't exist which it does and the ENGDATA[0] label doesn't exist. I suspect that is why there is no test example code with nitro. If someone has a simple example of a successfull ENGRDA addition please post.

**Comment by bradh** [comment]
Perhaps you can post a minimal, self-contained, compileable example of what you are trying to do?

**Reply by sherrick01821** [reply]
nitf::TRE engrda("ENGRDA","ENGRDA");

float mydata = 100.f;

engrda.setField("RECNT",1);
engrda.setField("ENGLN[0]",6);
engrda.setField("ENGLBL[0]","mydata");
engrda.setField("ENGMTXC[0]",1);
engrda.setField("ENGMTXR[0]",1);
engrda.setField("ENGDTS[0]",4);
engrda.setField("ENGDATC[0]",1);
engrda.setField("ENGDATA[0]",mydata);

**Reply by bradh** [reply]
Sorry I wasn't clearer. Try reading http://sscce.org/ 

Minimise the work everyone else has to do to help you. Please provide a  complete compilable sample that demonstrates the issue.

**Comment by sherrick01821** [comment]
#include <iostream>
#include <nitf/TRE.hpp>
#include <except/Exception.h>


int main()
{

#if 0
  nitf::TRE tre( "PIXMTA" , "PIXMTA" );

  tre.setField("NUMAIS", "001");

#if 1
  nitf::TREFieldIterator it = tre.begin();
  while( it != tre.end() )
  {
      	nitf::Pair pair = *it;
       	std::cout <<  pair.first() << " " << pair.second() << std::endl;

     	++it;
  }
#endif



  tre.setField("AISDLVL[0]", 200, true);

#endif

  nitf::TRE engrda("ENGRDA" , "ENGRDA" );

  engrda.setField("RESRC","HSS");
  engrda.setField("RECNT",1);

  nitf::TREFieldIterator it = engrda.begin();
  while( it != engrda.end() )
  {
      	nitf::Pair pair = *it;
       	std::cout <<  pair.first() << std::endl;

     	++it;
  }

  try {
  char val = 'A';
  engrda.setField("ENGLN[0]",5);
  engrda.setField("ENGMTXC[0]",1);
  engrda.setField("ENGMTXR[0]",1);
  engrda.setField("ENGTYP[0]","A");
  engrda.setField("ENGDTS[0]",1);
  engrda.setField("ENGDATC[0]",1);
  engrda.setField("ENGDATA[0]",val);
  }
  catch( except::Exception & e )
  {
    std::cout << "Threw exception " <<  e.what() << std::endl;
  } 


}


This example throws an exception that ENGDATA[0] is not a recognixed field for this TRE 

the fields printed in the iterator loop are
RESRC
RECNT
ENGLN[0]
ENGMTXC[0]
ENGMTXR[0]
ENGTYP[0]
ENGDTS[0]
ENGDATU[0]
ENGDATC[0]

Can someone run this and see if they get the same issue? Why isn't ENGDATA[0] not a valid field and how does one put a value in?

**Comment by sherrick01821** [comment]
Then how do you set the actual engineering data? ENGDATA is a valid field according to the nga documentation 


Sent from Yahoo Mail on Android 
 
  On Wed, Aug 25, 2021 at 5:55 PM, J. Daniel ***@***.***> wrote:   


ENGDATA[0] is not a field; see test_tre_mods++.cpp

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by JDanielSmith** [reply]
In the [bugfix/ENGRDA](https://github.com/mdaus/nitro/tree/bugfix/ENGRDA) branch:
```
TEST_CASE(basicIteration_ENGRDA)
{
    nitf::TRE engrda("ENGRDA", "ENGRDA");

    engrda.setField("RESRC", "HSS");
    engrda.setField("RECNT", 1, true /*forceUpdate*/);

    engrda.setField("ENGDTS[0]", 8); // size
    engrda.setField("ENGDATC[0]", 1); // count
    engrda.updateFields();
    engrda.setField("ENGDATA[0]", "ABC");
}
```

**Comment by sherrick01821** [comment]
The TRE is ENGRD but the field I want to set is called ENGDATA
this is an example I found online



    On Wednesday, August 25, 2021, 8:30:21 PM EDT, Steven Herrick ***@***.***> wrote:  
 
 Then how do you set the actual engineering data? ENGDATA is a valid field according to the nga documentation 


Sent from Yahoo Mail on Android 
 
  On Wed, Aug 25, 2021 at 5:55 PM, J. Daniel ***@***.***> wrote:   


ENGDATA[0] is not a field; see test_tre_mods++.cpp

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by sherrick01821** [comment]
Interesting, the test mod code you pointed to has the ENGDATA commentred out, probably because it doesn't work



    On Wednesday, August 25, 2021, 8:30:21 PM EDT, Steven Herrick ***@***.***> wrote:  
 
 Then how do you set the actual engineering data? ENGDATA is a valid field according to the nga documentation 


Sent from Yahoo Mail on Android 
 
  On Wed, Aug 25, 2021 at 5:55 PM, J. Daniel ***@***.***> wrote:   


ENGDATA[0] is not a field; see test_tre_mods++.cpp

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by sherrick01821** [comment]
I'll try it today, thanks

    On Thursday, August 26, 2021, 10:34:10 AM EDT, J. Daniel Smith ***@***.***> wrote:  
 
 

TEST_CASE(basicIteration_ENGRDA)
{
    nitf::TRE engrda("ENGRDA", "ENGRDA");

    engrda.setField("RESRC", "HSS");
    engrda.setField("RECNT", 1, true /*forceUpdate*/);

    engrda.setField("ENGDTS[0]", 8); // size
    engrda.setField("ENGDATC[0]", 1); // count
    engrda.updateFields();
    engrda.setField("ENGDATA[0]", "ABC");
}

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by sherrick01821** [comment]
Do you know how to put an array say on integrers in as ENGDATA? I knwo I would use ENGMTXC and ENGMTXR buit I 'm not sure uif I just put the array as the arg to ENGDATA[0]

    On Thursday, August 26, 2021, 10:34:10 AM EDT, J. Daniel Smith ***@***.***> wrote:  
 
 

TEST_CASE(basicIteration_ENGRDA)
{
    nitf::TRE engrda("ENGRDA", "ENGRDA");

    engrda.setField("RESRC", "HSS");
    engrda.setField("RECNT", 1, true /*forceUpdate*/);

    engrda.setField("ENGDTS[0]", 8); // size
    engrda.setField("ENGDATC[0]", 1); // count
    engrda.updateFields();
    engrda.setField("ENGDATA[0]", "ABC");
}

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
